packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -3080,18 +3081,22 @@ export class ContainerRuntime
 		address: string | undefined,
 		type: string,
 		content: any,
+		targetClientId?: string,
 	): ISignalEnvelope {
 		const newSequenceNumber = ++this._perfSignalData.signalSequenceNumber;
 		const newEnvelope: ISignalEnvelope = {
 			address,
 			clientSignalSequenceNumber: newSequenceNumber,
 			contents: { type, content },
+			targetClientId,
 		};
 
 		// We should not track any signals in case we already have a tracking number.
+		// We should also only track signals that we expect to return to the current client.
 		if (
 			newSequenceNumber % this.defaultTelemetrySignalSampleCount === 1 &&
-			this._perfSignalData.trackingSignalSequenceNumber === undefined
+			this._perfSignalData.trackingSignalSequenceNumber === undefined &&
+			(targetClientId === undefined || targetClientId === this.clientId)
```

Comment:
I think we already don't have enough of these samples and I don't think we want less (and in the degenerate case none). Can we change up the tracking to catch more drops? (This is a big question for us.)
And for performance sampling, how do we alter the mechanism to allow for more that won't come back? An example of possible solution to have more tracking is to adjust the remainder check when we skip. Then the next could be given an opportunity to check.

Reply:
I adjusted the defaultTelemtrySignalSampleCount to catch 10x more signals. 

re: tracking drops that won't come back -- I think this would either involve some sort of ack mechanism or centralized unified signal sequence number on the service, similar to what we do for ops. I'm not sure if this is desirable but these are the options I have so far.

Reply:
10x doesn't actually mean we avoid degenerate cases.
Is there some history to the original frequency? It may not be reasonable to increase the rate. (I don't know why it is throttled at all, which means I don't know the impact of changing the rate.)
Definitely not looking for ack - I think any acknowledgement is a higher level bit of logic.

Reply:
What's meant by degenerate cases here? The extra condition step added here is just to track both broadcast signals and self-targeted signals which catches more drops is more than we're currently doing.

Reply:
Changed to not include targeted signals in the current tracking mechanism.